hypertrace-trace-enricher/hypertrace-trace-enricher/src/main/java/org/hypertrace/traceenricher/trace/enricher/StructuredTraceEnrichProcessor.java

```diff
@@ -33,14 +33,14 @@
 
   private static final Logger logger = LoggerFactory.getLogger(StructuredTraceEnrichProcessor.class);
   private static EnrichmentProcessor processor = null;
+  private DefaultClientRegistry clientRegistry;
 
   private static final String ENRICHED_TRACES_COUNTER = "hypertrace.enriched.traces";
   private static final ConcurrentMap<String, Counter> tenantToEnrichedTraceCounter = new ConcurrentHashMap<>();
 
   private static final String ENRICHED_TRACES_TIMER = "hypertrace.trace.enrichment.latency";
   private static final ConcurrentMap<String, Timer> tenantToEnrichmentTraceTimer = new ConcurrentHashMap<>();
 
-
   @Override
   public void init(ProcessorContext context) {
     if (processor == null) {
```

Comment:
I'm not sure what the parallelism expectations are on this class. It reads like we expect init to be called from multiple threads, but we only want a single `EnrichmentProcessor` instance. I'm not sure there are statics being used, do we expect there to be multiple `StructuredTraceEnrichProcessor` instances sharing one `EnrichmentProcessor`?

The init side of this will handle that fine, but the cleanup side - which prior to this PR was ignored and not doing any cleanup - would be problematic. We'd have to use something like a countdown latch to ensure we don't clean up until all instances are out of use.

Reply:
Dug way back in the archives - looks like this was written back before this was written on kafka streams. So does someone with more context on kafka streams know if it's still needed?

Reply:
@surajpuvvada @buchi-busireddy

Reply:
Yes there can be multiple instances. Typically you can imagine one per stream thread. I think the EnrichmentProcessor might be using a cache somewhere and my guess is it is to prevent duplicate caches across instances. I vaguely recall running into memory issues..

Reply:
So there are multiple instances of `StructuredTraceEnrichProcessor` all sharing the same underlying instance of `EnrichmentProcessor`? What could go wrong 🙄 !

Given that, I'll remove the cleanup happening now (which is bad, but mirrors the previous implementation). I think separately, it's worth allowing the creation of multiple instances here, since the client registry allows sharing caches across instances now.